### PR TITLE
Remove Loadbalancer interface and use k8s-cloud-provider mocks

### PR DESCRIPTION
**Pull request by spencerhance**
This requires some new common hooks in k8s-cloud-provider, PR: https://github.com/GoogleCloudPlatform/k8s-cloud-provider/pull/16 and pulling in those changes to vendor

Used https://github.com/kubernetes/ingress-gce/pull/340 as a reference.

### CODE REVIEW

File: pkg/loadbalancers/loadbalancers_test.go [hunk 1]
```diff
@@ -46,6 +53,79 @@ var (
...
+	pool    LoadBalancerPool
+	fakeGCE *gce.Cloud
+	namer   *utils.Namer
+	t       *testing.T
+}
+
+func newTestJig(t *testing.T) *testJig {
+	namer := utils.NewNamer(clusterName, "fw1")
+	fakeGCE := gce.NewFakeGCECloud(gce.DefaultTestClusterValues())
+
```
Comment (bowei) [inline pkg/loadbalancers/loadbalancers_test.go]: mock := fakeGCE.Compute().(*cloud.MockGCE)

makes the code below less verbose

Comment (bowei) [reply pkg/loadbalancers/loadbalancers_test.go]: or better yet, keep a pointer to mock in testJig (e.g. `j.mock`) as it is used all the time

**Comment by rramkumar1** [comment]
@spencerhance do you mind running a code coverage tool before & after just to verify there was no loss in coverage? It's not really a foolproof method to sanity check we are not losing coverage but I can't think of anything else.

File: pkg/loadbalancers/loadbalancers_test.go (hunk 1 above)
Comment (spencerhance) [reply pkg/loadbalancers/loadbalancers_test.go]: done

**Comment by spencerhance** [comment]
@rramkumar1 `go test -cover` reports pkg/loadbalancers went from 75.9% to 74.1%

**Comment by bowei** [comment]
need to revert until the vendor stuff is fixed

**Pull request merged**